test/distributed/fsdp/test_fsdp_fine_tune.py

```diff
@@ -276,6 +299,86 @@ def _test_parity_with_ddp(
             torch.testing.assert_close(losses[0], losses[1])
             losses.clear()
 
+    @skip_if_lt_x_gpu(2)
+    def test_parity_with_non_frozon_fsdp(self):
+        """
+        For frozon modules with unused input, reshard could happen without unshard
```

Comment:
I am curious why a module would have an unused input. Is it because the module just forwards the input to the output? (In the unit test's `LinearUnusedInput`, it seems more like the module does not need that input at all.)

(I am not saying we should not fix this. I am just curious why we would have this case.)

Reply:
there are ``if-else`` in forward and `input2` becomes unused in following examples

```
if isinstance(layer, some_class):
    layer(input1, input2)
else:
    layer(input1)
```

Reply:
Got it! I recall this in internal code :)

In particular, I recall that `isinstance(layer, some_class)` broke with FSDP because FSDP changed the module's class due to being a wrapper.

Reply:
exactly. we are probably looking at the same code